api/src/main/java/io/grpc/ClientStreamTracer.java

```diff
@@ -79,6 +79,13 @@ public void inboundHeaders() {
   public void inboundTrailers(Metadata trailers) {
   }
 
+  /**
+   * Information providing context to the call became available.
+   */
+  @Internal
+  public void addOptionalLabel(String key, String value) {
```

Comment:
We talked about it earlier but is it possible to do anything smarter here to avoid string comparisons on the key?

Reply:
There's not many options. You hard-coded it for C, so it isn't much to go on.

Literal strings are intern'd, so if they are equal, it would only take a reference check. For non-equals, hashCode hopefully helps avoid the character iteration.

We could require a global registry for optional label keys, that would make it always reference checks, but that difference may not matter.

I think the bigger problem is we have to check whether the optional key is included in the metric every RPC, and that means digging into maps and such and dynamic allocation.